### DISCUSSION TITLE
Update rows automation return with "invalid inputs" when fields are left blank

### DISCUSSION THREAD

**Discussion by Bergem93** [Budibase/budibase#9744] [Help] [answered]
Hi,

I'm trying to run an automation which checks assets purchase date and flags all assets which where bough more than 2,5 years ago. 

First i'm quering all rows in assets with purchase date < (todays date - 2,5 years). That works and returns the expected rows.

Next i'm trying to update the "Less than 6 months" boolean value to TRUE, and all other columns should be left untouched, However, i'm just getting and error on the third step with "Invalid inputs". If i try to fill in all other columns aswell, i get the error "row does not exist". 

![Automation1](https://user-images.githubusercontent.com/5398191/220079785-10736fdc-e45d-4119-9c76-af31dfc34134.png)
![Automation2](https://user-images.githubusercontent.com/5398191/220079805-4eccff4a-6d4d-432f-b3f8-d46e20a1011a.png)

Filter to query rows. Works as intended.
![Query rows filter](https://user-images.githubusercontent.com/5398191/220079903-411f7b0b-3be3-42dd-9431-6470554aa5a5.png)

Test details after running with only "Less than 6 months" set to TRUE. 
![Test details](https://user-images.githubusercontent.com/5398191/220080028-fbe5def2-14e4-4f9a-af96-0f8fa291addf.png)

Any pointers? Am I missing something obvious?

**Comment by andz-bb** [comment, marked as answer]
hey @Bergem93 - thanks for your detailed question

For this it looks like you need to loop the Update Row action over the results from the Query Rows step. To do this, select _Add Looping_ at the top of your Update Row action, and set it to use input type Array. Then set the Binding/Value to be `{{ steps.1.rows }}` - this means that the Update Row will run for each result from the Query Rows step.

Finally, you just need to provide the relevant _id for each row that you want to update in that action. Because this is running in a loop, you can use the binding for the item that is currently being used. In your case it will be `{{ loop.currentItem._id }}` for the Row ID field at the bottom.

Your config will look something like this:
<img width="1028" alt="automation config" src="https://user-images.githubusercontent.com/110921612/220106481-2841f18c-2c2a-445b-9e91-60bba8149a2c.png">

If you need me to explain in any more detail let me know! Hope this helps :)

**Reply by Bergem93** [reply]
Hi @andz-bb,

Thank you for a very good answer! It all makes sense now, but still having some issues.
I tried setting up the loop as explained, but when running the test, it gets stuck here. 

Just spins infinitely. Does not work to click cancel either. Tried setting the Max loop iterations to 10, same result. 
![image](https://user-images.githubusercontent.com/5398191/220114086-a01d053a-2077-4f8c-a118-61d3001088d2.png)


Any ideas?

**Reply by andz-bb** [reply]
hey @Bergem93 - could you try opening up your browser's developer tools (F12 in Chrome) and then refreshing/running the test again? There might be an error in the Console or the Network tab that gives us a better idea of what's going wrong. Alternatively feel free to send an app export to support@budibase.com if you're able to, and I can have a closer look on my end 👍

**Reply by Bergem93** [reply]
Hi @andz-bb 

I get this error in the Console.

![image](https://user-images.githubusercontent.com/5398191/220132559-01b78a3c-40a3-41f6-8787-6d848d6ce489.png)

Network log
![image](https://user-images.githubusercontent.com/5398191/220132904-b81d3012-e330-4671-b3f0-7cecaf5afb38.png)

**Reply by andz-bb** [reply]
@Bergem93 thanks for the quick reply! Would it be possible for you to send over an app export so I can do some debugging?

**Reply by Bergem93** [reply]
@andz-bb I've tried sending an export til support@budibase.com, but the email gets rejected. Any other ways to send an export?

**Reply by andz-bb** [reply]
@Bergem93 Sorry for the inconvenience - some email providers don't like the format of our app exports! If you have Discord you should be able to send the app export to me there. I'm user `Andrew T#0001`. [Here's an invite to our server if you would like to join.](https://discord.gg/ZepTmGbtfF) Alternatively, you could upload the app export to a file sharing service and then send the link by email.

**Reply by andz-bb** [reply]
@Bergem93 thanks for sending the app export! I found a few anomalies with your schema in the Assets table that were causing the Update Row automation step to fail. There are a number of Formula columns but whenever I imported the app I noticed that for the Age and Current Value columns the Formula type was unset. 

I'm not entirely sure how this could have happened but I set them both to dynamic. After that I deleted the looped Update Row action from the Automation and re-added it to make sure it had the updated schema for the Assets table. 

<img width="1512" alt="Bergem automation fix" src="https://user-images.githubusercontent.com/110921612/220178701-e0e80931-cb9d-4683-949a-e84d3a899986.png">

Now the automation runs and applies the update to two of the rows in the table! If you need my updated app export or any further guidance on this let me know and I'll be happy to help 😊

**Reply by Bergem93** [reply]
Thank you so much @andz-bb! Now it works as intended :D
